### PR TITLE
Added solution for leetcode 50- Pow(x,n)!

### PR DESCRIPTION
**Pull request by debangi29**
## Related Issue

#811 

## Description

[Please include a brief description of the changes or features added]
Added Leetcode Solution to problem 50 - Pow (x,n)

## Type of PR

- [x] Feature enhancement

## Screenshots / Videos (if applicable)
image

[Attach any relevant screenshots or videos demonstrating the changes]

## Checklist
- [x] I have performed a self-review of my code.
- [x] I have read and followed the Contribution Guidelines.
- [x] I have tested the changes thoroughly before submitting this pull request.
- [x] I have provided relevant issue numbers, screenshots, and videos after making the changes.
- [x] I have commented my code, particularly in hard-to-understand areas.

## Additional Context

[Include any additional information or context that might be helpful for reviewers.]

## Resources for Guidance

Here are some resources that may be helpful as you contribute to CodeHarborHub:
- [Code Harbor Hub Community Features](https://www.codeharborhub.live/community/features)
- [Docusaurus Documentation](https://docusaurus.io/docs/create-doc)
- [React.js Documentation](https://legacy.reactjs.org/docs/getting-started.html)
- [Markdown Guide](https://www.markdownguide.org/)
- [MDX Documentation](https://mdxjs.com/docs/)
- [Mermaid Documentation](https://mermaid.js.org/)

**Resources for DSA Features on Our Site:**

- Tabs: [Switch Languages](https://www.codeharborhub.live/community/features#multi-language-support-code-blocks)
- [Line Numbering](https://www.codeharborhub.live/community/features#line-numbering)
- [Math Equations](https://www.codeharborhub.live/community/features#math-equations-with-katex)

### CODE REVIEW

File: dsa-solutions/lc-solutions/0000-0099/0050-pow(x,n).md [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+
+## Problem Description
+
+| Problem Statement | Solution Link | LeetCode Profile |
+| :---------------- | :------------ | :--------------- |
+| [Pow(x,n)](https://leetcode.com/problems/powx-n/description/) | [Pow(x,n) Solution on LeetCode](https://leetcode.com/problems/powx-n/solutions/) |  [Leetcode Profile](https://leetcode.com/u/debangi_29/) |
+
+## Problem Description
+
+Implement pow(x, n), which calculates x raised to the power n (i.e., x<sup>n</sup>).
```
Comment (Ajay-Dhangar) [inline dsa-solutions/lc-solutions/0000-0099/0050-pow(x,n).md:19]: replace `x<sup>n</sup>` to `$x^n$`

File: dsa-solutions/lc-solutions/0000-0099/0050-pow(x,n).md [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+**Input**: x = 2.00000, n = -2
+
+**Output**: 0.25000
+
+**Explanation**: 2-2 = 1/22 = 1/4 = 0.25
+ 
+
+### Constraints
+
+- -100.0 < x < 100.0
```
Comment (Ajay-Dhangar) [inline dsa-solutions/lc-solutions/0000-0099/0050-pow(x,n).md:49]: ### Bad practice

```md

- -100.0 < x < 100.0
- -2<sup>31</sup> <= n <= 2<sup>31</sup>-1
- n is an integer.
- Either x is not zero or n > 0.
- -10<sup>4</sup> <= x<sup>n</sup> <= 10<sup>4</sup>
```

### good practice

```md
- $-100.0 < x < 100.0$
- $-2^{31} \leq n \leq 2^{31} - 1$
- $n$ is an integer.
- Either $x$ is not zero or $n > 0$.
- $-10^{4} \leq x^{n} \leq 10^{4}$
```

**Review by Ajay-Dhangar — CHANGES_REQUESTED** [review]
update your code based on suggestions

**Comment by debangi29** [comment]
I have made the requested changes @Ajay-Dhangar

**Review by Ajay-Dhangar — APPROVED** [review]
Done

**Pull request merged**